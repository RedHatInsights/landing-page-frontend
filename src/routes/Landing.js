import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';
import { withRouter } from 'react-router-dom';
import {
    Modal,
    Button,
    Title,
    Stack,
    StackItem
} from '@patternfly/react-core';

import Header from '../layout/Header';
import Body from '../layout/Body';
import Marketing from '../layout/Marketing';
import FooterTraditional from '../layout/FooterTraditional';
import { activeTechnologies } from '../consts';
import './Landing.scss';

class Landing extends Component {
    state = {
        isModalOpen: false
    }

    componentDidMount() {
        const { location } = this.props;

        const params = location.search.slice(1).split('&').reduce((acc, curr) => ({
            ...acc,
            [curr.split('=')[0]]: Object.values(activeTechnologies).find(item => item.entitlement === curr.split('=')[1])
        }), {});

        this.setState({
            ...params,
            isModalOpen: params && Object.keys(params).length > 0
        });

        window.insights.chrome.auth.getUser().then(user => {
            if (user) {
                this.setState({ unauthed: false });
            } else {
                this.setState({ unauthed: true });
            }
        }).catch(() => {
            this.setState({ unauthed: true });
        });

    }

    handleModalToggle = () => {
        this.setState({ isModalOpen: false });
    }

    render() {
        const { isModalOpen, not_entitled: notEntitled, unauthed } = this.state;

        return (
            <Fragment>
                { unauthed
                    ? <Marketing />
                    : <Fragment>
                        <Header />
                        <Body />
                    </Fragment>
                }
                <FooterTraditional />
                { notEntitled && <Modal
                    title={ 'You are not entitled to use this application' }
                    className='ins-c-error-modal'
                    hideTitle={ true }
                    isOpen={ isModalOpen }
                    onClose={ this.handleModalToggle }
                >
                    <Stack gutter='md' className='ins-c-error-state'>
                        <StackItem className='ins-c-error-state__title'>
                            <Title headingLevel="h3" size="2xl">{ notEntitled.emptyTitle }</Title>
                        </StackItem>
                        <StackItem className='ins-c-error-state__image'>
                            { notEntitled.icon && <notEntitled.icon
                                className="ins-c-icon__active"
                                aria-hidden
                                alt={ `${notEntitled.title} logo` }
                                { ...notEntitled.iconProps }
                            /> }
                            { notEntitled.image && <img
                                className="ins-c-application-info__logo"
                                aria-hidden
                                src={ notEntitled.image }
                                alt={ `${notEntitled.title} logo` } /> }
                        </StackItem>
                        <StackItem className='ins-c-error-state__body'>
                            { notEntitled.emptyText }
                        </StackItem>
                        <StackItem className='ins-c-error-state__footer'>
                            {
                                notEntitled.emptyAction.title &&
                                    <Button variant="primary" className='ins-c-error-state__footer-action' onClick={ () => {
                                        if (notEntitled.emptyAction.navigate) {
                                            window.location.href = notEntitled.emptyAction.navigate;
                                        }
                                    } } >
                                        { notEntitled.emptyAction.title }
                                    </Button>
                            }
                            <Button variant="link" className='ins-c-error-state__footer-close' onClick={ this.handleModalToggle }>
                                { notEntitled.emptyAction.close ? `${notEntitled.emptyAction.close }` : 'Close' }
                            </Button>
                        </StackItem>
                    </Stack>
                </Modal> }
            </Fragment>
        );
    }
}

Landing.propTypes = {
    history: PropTypes.object,
    loadTechnologies: PropTypes.func,
    location: PropTypes.shape({
        search: PropTypes.string
    })
};

Landing.defaultProps = {
    loadTechnologies: () => undefined
};

export default withRouter(Landing);
