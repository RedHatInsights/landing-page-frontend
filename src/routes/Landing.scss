@import '~@redhat-cloud-services/frontend-components-utilities/styles/_mixins';
@import '~@patternfly/patternfly/sass-utilities/_all';

:root {
    --ins-global--Color--green: #37912f;
}

.ins-c-application-info .ins-m-gray { color: var(--pf-global--Color--200); }

.ins-m-tech-preview { color: var(--ins-global--Color--green); }

.ins-m-dev-preview { color: #39A5DC; }

.ins-m-early-preview {
    width: max-content;
    color: var(--pf-global--default-color--200);
}

.pf-c-modal-box__body .pf-c-empty-state img.ins-c-application-info__logo {
    @include rem('height', 76px);
}

.pf-c-card {

    &.ins-c-application-info {
        height: 100%;

        img.ins-c-application-info__logo {
            @include rem('height', 56px);
            max-width: initial;
        }

        svg.ins-c-application-info__logo {
            @include rem('height', 56px);
            @include rem('width', 56px);
        }

        h2 { margin-top: 15px; }
    }

    .ins-c-application-info__open {
        display: block;
        span { margin-right: 15px }
    }

    .ins-c-application-info__content-applist {
        display: flex;
        flex-direction: column;

        a {
            display: block;
            text-transform: capitalize;
        }

        a + a { margin-top: 10px; }
    }

    .ins-c-navigation { color: var(--pf-global--active-color--100); }
}

svg.ins-c-icon__active { fill: var(--pf-global--active-color--100); }

.pf-c-modal-box .pf-c-title { white-space: normal; }

sup.ins-c-rball { @include rem('font-size', 10px); }

.ins-c-application-info__split {
    height: 100%;
    .ins-c-application-info__open {
        margin-top: auto;
        margin-bottom: 0;
    }
}

// Unentitled Modal Styles
.ins-c-error-modal {
    &__header {
        text-align: center;
        max-width: 75%;
        margin-left: auto;
        margin-right: auto;
        padding-bottom: var(--pf-global--spacer--md);
    }
    &__content {
        align-items: center;
        justify-content: center;
        text-align: center;
        &--body { width: 75%; }
    }
}

// Alert message
.notifications-portal .pf-c-alert{
    .pf-c-alert__description { display: none; }
    .pf-c-alert__title {
        padding: unset;
    }

    .pf-c-alert__icon {
        padding: unset;
        background-color: unset;
    }
}

// Adjust PF layout to be able to display without header
.ins-c-page__landing-layout {
    height: 100%;

    .ins-c-page__landing-navigation {
        flex-shrink: 0;
    }
}

.ins-c-page__landing-content {
    width: 100%
}
