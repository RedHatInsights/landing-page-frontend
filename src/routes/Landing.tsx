import React, { Fragment } from 'react';
import { useFlag } from '@unleash/proxy-client-react';

import '../components/app-content-renderer/styles/panels.scss';

import FirstPanel from '../components/app-content-renderer/first-panel';
import SecondPanel from '../components/app-content-renderer/second-panel';
import VirtualAssistant from '../components/app-content-renderer/virtual-assistant';
import { useLoadModule } from '@scalprum/react-core';
import { PageSection } from '@patternfly/react-core/dist/dynamic/components/Page';

const getWidgetLayoutLandingPage = () => {
  const scope = 'widgetLayout';
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: GridLayout }] = useLoadModule(
    { scope, module: './WidgetLayout' },
    {}
  );
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: AddWidgetDrawer }] = useLoadModule(
    { scope, module: './WidgetDrawer' },
    {}
  );
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  const [{ default: Header }] = useLoadModule(
    { scope, module: './WidgetHeader' },
    {}
  );
  if (!GridLayout || !AddWidgetDrawer || !Header) {
    return <></>;
  }
  return (
    <>
      <Header />
      <AddWidgetDrawer dismissible={false}>
        <PageSection>
          <GridLayout isLayoutLocked={false} />
        </PageSection>
      </AddWidgetDrawer>
    </>
  );
};

const Landing = () => {
  const widgetLayoutLandingPageEnabled = useFlag(
    'platform.landing-page.widgetization'
  );
  return (
    <div className="land-c-page-content pf-v5-u-display-flex pf-v5-u-flex-direction-column">
      <Fragment>
        <VirtualAssistant />
        {widgetLayoutLandingPageEnabled ? null : <FirstPanel />}
        {widgetLayoutLandingPageEnabled ? null : <SecondPanel />}
        {widgetLayoutLandingPageEnabled ? getWidgetLayoutLandingPage() : null}
      </Fragment>
    </div>
  );
};

export default Landing;
