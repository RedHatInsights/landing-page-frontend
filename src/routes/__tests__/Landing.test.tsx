/* eslint-disable react/prop-types */
import React from 'react';
import { act, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ScalprumProvider } from '@scalprum/react-core';
import Landing from '../Landing';
import FlagProvider, { UnleashClient } from '@unleash/proxy-client-react';

const LandingWrapper: React.FC<React.PropsWithChildren> = ({ children }) => {
  const unleashClient = React.useMemo(
    () =>
      new UnleashClient({
        url: `${document.location.origin}/api/featureflags/v0`,
        clientKey: 'proxy-123',
        appName: 'web',
        fetch: () =>
          Promise.resolve({
            status: 200,
            body: {
              toggles: [],
            },
          }),
      }),
    []
  );
  return (
    <ScalprumProvider
      config={{}}
      api={{
        chrome: {
          getEnvironment: () => '',
          isProd: () => false,
          isBeta: () => false,
        },
      }}
    >
      <FlagProvider unleashClient={unleashClient}>
        <MemoryRouter>{children}</MemoryRouter>
      </FlagProvider>
    </ScalprumProvider>
  );
};

describe('<Landing/>', () => {
  test('should render correctly', async () => {
    let container;
    await act(async () => {
      const { container: c } = render(
        <LandingWrapper>
          <Landing />
        </LandingWrapper>
      );
      container = c;
    });
    expect(container).toMatchSnapshot();
  });
});
