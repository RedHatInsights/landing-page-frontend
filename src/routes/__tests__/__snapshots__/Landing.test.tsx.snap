// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing/> should render correctly 1`] = `
<div>
  Scalprum Component Mock
</div>
`;
